File: runinator-command-center/src/utils/inputs.ts
import type { WorkflowNodeRun } from "../types/models";
import { displayValue } from "./values";

export function isInputWaitingStatus(status: unknown): boolean {
  return ["waiting", "input_required", "pending"].includes(normalizeStatus(status));
}

export function inputValueFromNodeRun(nodeRun: WorkflowNodeRun): unknown {
  return nodeRun.output_json ?? nodeRun.state?.input ?? null;
}

function normalizeStatus(status: unknown): string {
  return displayValue(status)
    .trim()
    .toLowerCase()
    .replaceAll("-", "_");
}

File: runinator-command-center-flutter/lib/core/utils/inputs.dart
// port of core/utils/inputs.ts.

import '../domain/models/index.dart';
import 'values.dart';

bool isInputWaitingStatus(Object? status) =>
    ['waiting', 'input_required', 'pending'].contains(_normalizeStatus(status));

Object? inputValueFromNodeRun(WorkflowNodeRun nodeRun) =>
    nodeRun.outputJson ?? nodeRun.state?['input'];

String _normalizeStatus(Object? status) => displayValue(status).trim().toLowerCase().replaceAll('-', '_');
